libvast/src/system/local_segment_store.cpp

```diff
@@ -0,0 +1,336 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/system/local_segment_store.hpp"
+
+#include "vast/detail/overload.hpp"
+#include "vast/detail/zip_iterator.hpp"
+#include "vast/fbs/partition.hpp"
+#include "vast/fbs/utils.hpp"
+#include "vast/ids.hpp"
+#include "vast/logger.hpp"
+#include "vast/plugin.hpp"
+#include "vast/query.hpp"
+#include "vast/segment_store.hpp"
+#include "vast/span.hpp"
+#include "vast/system/node_control.hpp"
+#include "vast/table_slice.hpp"
+
+#include <caf/settings.hpp>
+#include <flatbuffers/flatbuffers.h>
+#include <fmt/format.h>
+
+#include <vector>
+
+namespace vast::system {
+
+namespace {
+
+// Handler for `vast::query` that is shared between active and passive stores.
+// Precondition: Query type is either `count` or `extract`.
+template <typename Actor>
+caf::result<atom::done>
+handle_lookup(Actor& self, const vast::query& query, const vast::ids& ids,
+              const std::vector<table_slice>& slices) {
+  std::vector<expression> checkers;
+  for (const auto& slice : slices) {
+    if (query.expr == expression{}) {
+      checkers.emplace_back();
+    } else {
+      auto c = tailor(query.expr, slice.layout());
+      if (!c)
+        return c.error();
+      checkers.emplace_back(prune_meta_predicates(std::move(*c)));
+    }
+  }
```

Comment:
If I understand it right the first loop exists to make an early exit in case one of the calls to `tailor` fails, but I think we should skip only those slices instead.

Reply:
The root cause of this is us currently lacking a way to return "partial success" from a query. If we just skip, then results might be silently missing without error notification for the user. This behavior is the same as the old (non-local) segment store,